### PR TITLE
Only allow outgoing req/res to be sent once

### PR DESCRIPTION
**Pull request by jcorbin**

### CODE REVIEW

File: node/reqres.js [hunk 1]
```diff
@@ -122,18 +130,25 @@ function TChannelOutgoingResponse(id, options, sendFrame) {
...
 inherits(TChannelOutgoingResponse, EventEmitter);
 
 TChannelOutgoingResponse.prototype.send = function send(err, res1, res2) {
     var self = this;
-    if (err) {
-        self.ok = false;
-        var errArg = isError(err) ? err.message : JSON.stringify(err); // TODO: better
-        self.sendFrame(self.name, res1, errArg);
+    if (self.sent) {
+        self.emit('error', new Error('response already sent'));
```
Comment (kriskowal) [inline node/reqres.js]: Does this constitute a programming error or an operational error?

File: node/reqres.js [hunk 2]
```diff
@@ -122,18 +130,25 @@ function TChannelOutgoingResponse(id, options, sendFrame) {
...
     } else {
-        self.sendFrame(self.name, res1, res2);
+        self.sent = true;
+        if (err) {
+            self.ok = false;
+            var errArg = isError(err) ? err.message : JSON.stringify(err); // TODO: better
```
Comment (kriskowal) [inline node/reqres.js]: For reference, #135

Comment (kriskowal) [inline node/reqres.js]: There is an Error frame type. Is that intended for this case?

Comment (jcorbin) [inline node/reqres.js]: no this is application error only, the error frame is protocol level error exclusively.

**Comment by kriskowal** [comment]
Looks good to me.

**Pull request merged**